protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/impl/StandardFieldImportResolver.kt

```diff
@@ -0,0 +1,132 @@
+import com.toasttab.protokt.rt.PType.FIXED64
+import com.toasttab.protokt.rt.PType.INT32
+import com.toasttab.protokt.rt.PType.INT64
+import com.toasttab.protokt.rt.PType.SFIXED32
+import com.toasttab.protokt.rt.PType.SFIXED64
+import com.toasttab.protokt.rt.PType.SINT32
+import com.toasttab.protokt.rt.PType.SINT64
+import com.toasttab.protokt.rt.PType.UINT32
+import com.toasttab.protokt.rt.PType.UINT64
+import com.toasttab.protokt.rt.SFixed32
+import com.toasttab.protokt.rt.SFixed64
+import com.toasttab.protokt.rt.SInt32
+import com.toasttab.protokt.rt.SInt64
+import com.toasttab.protokt.rt.UInt32
+import com.toasttab.protokt.rt.UInt64
+import kotlin.reflect.KClass
+
+class StandardFieldImportResolver(
+    private val f: StandardField,
+    private val ctx: PluginContext,
+    private val pkg: PPackage
+) {
+    fun imports(): ImmutableSet<Import> =
+        listImports() +
+            mapImports() +
+            ptypeImports() +
+            wrapperTypeImports()
+
+    private fun listImports() =
+        if (f.repeated) {
+            immutableSetOf(
+                rtMethod("copyList"),
+                rtMethod("finishList")
+            ) +
+                if (f.packed) {
+                    immutableSetOf(pclass(UInt32::class))
+                } else {
+                    immutableSetOf()
+                }
+        } else {
+            immutableSetOf()
+        }
+
+    private fun mapImports() =
+        if (f.map) {
+            setOf(
+                rtMethod("sizeofMap"),
+                rtMethod("finishList")
+            )
+        } else {
+            setOf()
+        }
+
+    private fun ptypeImports() =
+        listOfNotNull(
+            when (f.type) {
+                FIXED32 -> Fixed32::class
+                FIXED64 -> Fixed64::class
+                INT32 -> Int32::class
+                INT64 -> Int64::class
+                SFIXED32 -> SFixed32::class
+                SFIXED64 -> SFixed64::class
+                SINT32 -> SInt32::class
+                SINT64 -> SInt64::class
+                UINT32 -> UInt32::class
+                UINT64 -> UInt64::class
+                else -> null
+            }?.let { pclass(it) }
+        )
+
+    private fun wrapperTypeImports(): Set<Import> {
+        val set = mutableSetOf<Import>()
+
+        if (f.options.protokt.bytesSlice) {
+            set.add(pclass(BytesSlice::class))
+        } else {
+            set.add(Import.Class(f.typePClass(ctx)))
+        }
+
+        f.foldWrap(
+            pkg,
+            ctx,
+            { },
+            { wrapper, wrapped ->
+                if (wrapped == ByteArray::class) {
+                    set.add(pclass(Bytes::class))
+                }
+                addIfNotInPackage(set, wrapper)
+                addIfNotInPackage(set, converter(wrapper, wrapped, ctx)::class)
+            }
+        )
+
+        return set
+    }
+
+    private fun addIfNotInPackage(set: MutableSet<Import>, kclass: KClass<*>) {
```

Comment:
I don't like passing collections as parameters expecting side effects on them (not just because of the surrounding code's functional style) - can this be folded (pun intended) into the block above as a filter / flatMap combo?

Reply:
This check can actually be removed since we do a package check higher up in the resolution.